src/main/java/io/gravitee/repository/jdbc/management/JdbcAlertRepository.java

```diff
@@ -92,6 +93,31 @@ protected String getId(final AlertTrigger alert) {
         }
     }
 
+    @Override
+    public List<AlertTrigger> findByReferenceAndReferenceIds(final String referenceType, final List<String> referenceIds) throws TechnicalException {
+        LOGGER.debug("JdbcAlertRepository.findByReferenceAndReferenceIds({}, {})", referenceType, referenceIds);
+        try {
+            List<AlertTrigger> rows = jdbcTemplate.query(getOrm().getSelectAllSql() + " where reference_type = ? and reference_id in ( "
+                            + getOrm().buildInClause(referenceIds) + " )"
+                    , (PreparedStatement ps) -> {
+                        ps.setString(1, referenceType);
+                        getOrm().setArguments(ps, referenceIds, 2);
+                    }, getOrm().getRowMapper());
+
+            List<AlertTrigger> alertTriggers = new ArrayList<>();
+            for (AlertTrigger alertTrigger : rows) {
+                addEvents(alertTrigger);
+                alertTriggers.add(alertTrigger);
+            }
+
+            return alertTriggers;
```

Comment:
I think the following is doing the same 🤔 
```suggestion
            return rows.stream().peek(this::addEvents).collect(Collectors.toList());
```